source/includes/connection-snippets/x509-snippets/ruby-connection.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# begin x509 connection
+require 'mongo'
+
+client = Mongo::Client.new(
+  ['mongodb+srv://<cluster-url>/test'],
```

Comment:
When the client constructor is given an array, each array element must be a host address (not a uri). I filed https://jira.mongodb.org/browse/RUBY-2031 to improve our error reporting in this case but the code as provided here does not work.

Additionally, although it is permissible to provide both URI and Ruby options in the same call, I think the code will be easier to follow and reuse if it picked one of the two strategies (i.e. either use host address and Ruby options or provide all options in the URI).

Reply:
Additionally, I suggest that `/test` be removed here. 

In the original PR combining SCRAM and X509, it was never included for X509 examples. I also made the argument in https://github.com/mongodb/docs-ecosystem/pull/573#discussion_r350334451 that it was irrelevant for most drivers. Lastly, the corresponding X509-only Python PR (https://github.com/mongodb/docs-ecosystem/pull/584/files) does not include a default database name.

Reply:
Updated the PR to use uri syntax (aside from the DB specification) with tls* uri options. The one thing is that `/test` needs to be in the uri to be equivalent to the other driver examples.

@p-mongo 
@jmikola

Reply:
> The one thing is that /test needs to be in the uri to be equivalent to the other driver examples.

My previous argument in https://github.com/mongodb/docs-ecosystem/pull/573#discussion_r348462260 was that `/<db-name>` was not relevant for _any_ drivers. I would still advocate that it be removed from the SCRAM and X509 examples for all languages.

Reply:
Unfortunately, some drivers do throw errors without it, for instance in go I get the following error if the url is `mongodb+srv://balboaclusterm10-5sfvt.mongodb.net/authSource=$external&tlsCertificateKeyFile=%s&authMechanism=MONGODB-X509` : 

# error:
``` error parsing uri: authsource without username is invalid```

Reply:
This appears to be a bug in the Go driver then. https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#connection-string-options says:

> authSource

>    For GSSAPI and MONGODB-X509 authMechanisms the authSource defaults to $external